server/hotrod/src/main/scala/org/infinispan/server/hotrod/Decoder2x.scala

```diff
@@ -153,14 +143,26 @@ object Decoder2x extends AbstractVersionedDecoder with ServerConstants with Log
       (h.flag & f.id) == f.id
    }
 
-   private def readLifespanOrMaxIdle(buffer: ByteBuf, useDefault: Boolean): Int = {
-      val stream = readUnsignedInt(buffer)
-      if (stream <= 0) {
-         if (useDefault)
-            EXPIRATION_DEFAULT
-         else
-            EXPIRATION_NONE
-      } else stream
+   private def readLifespanMaxIdle(buffer: ByteBuf, usingDefaultLifespan: Boolean, usingDefaultMaxIdle: Boolean, version: Byte): (ExpirationParam, ExpirationParam) = {
+      def readDuration[T](isInvalid: T => Boolean, read: ByteBuf => T, defaultExpiration: => T = EXPIRATION_DEFAULT, noExpiration: => T = EXPIRATION_NONE)(useDefault: Boolean): T = {
+         val duration = read(buffer)
+         if (isInvalid(duration)) {
+            if (useDefault) defaultExpiration else noExpiration
+         } else duration
+      }
+      version match {
+         case VERSION_22 =>
```

Comment:
It seems maybe we should make a Decoder22.scala and have this in there instead and just override the method?